buildfix.sh

```diff
@@ -14,9 +63,52 @@ gofmt -w .
 
 if which clang-format &>/dev/null; then
   echo "Formatting .proto files..."
-  clang-format -i --style=Google $(git ls-files --exclude-standard | grep '\.proto$')
+  git ls-files --exclude-standard | grep '\.proto$' | xargs -d '\n' --no-run-if-empty clang-format -i --style=Google
 else
-  echo -e "${c_yellow}WARNING: Missing clang-format; will not format proto files.${c_reset}"
+  echo -e "${c_yellow}WARNING: Missing clang-format tool; will not format proto files.${c_reset}"
 fi
 
-echo "All Done!"
+echo "Formatting frontend and markup files..."
+bazel run --ui_event_filters=-info,-stdout,-stderr --noshow_progress //tools/prettier:fix
+
+if ((gazelle)); then
+  echo "Fixing BUILD dependencies with gazelle..."
+  bazel run --ui_event_filters=-info,-stdout,-stderr --noshow_progress //:gazelle
+fi
+
+git diff --name-only | sort >"$modified_after"
+
+echo 'All done!'
+
+if ! ((stage)); then
+  exit
+fi
+if ! [ -t 0 ] || ! [ -t 1 ]; then
+  echo -e "${c_yellow}WARNING: not connected to a terminal; skipping interactive staging.${c_reset}"
+  exit 1
+fi
+
+export PATH="$PATH:$HOME/go/bin"
+if ! which fzf &>/dev/null; then
+  echo -e "${c_yellow}WARNING: Missing fzf tool; did not stage any changes.${c_reset}"
+  exit 1
+fi
+
+# Only commit staged files if there were no files staged before running this script.
+prev_stage_clean=$(git diff --staged --quiet && echo 1 || echo 0)
+
+diff -Pdpru "$modified_before" "$modified_after" | perl -n -e '/^\+([^+].*)/ && print "./$1\n"' |
+  fzf --exit-0 --prompt='Stage modified files? Select with Arrows, Tab, Shift+Tab, Enter; quit with Ctrl+C > ' --multi |
+  xargs -d '\n' --no-run-if-empty git add
+
+if ((commit)) && ! git diff --staged --quiet; then
+  if ! ((prev_stage_clean)); then
+    echo -e "${c_yellow}WARNING: Files will need to be manually committed since there were existing staged files.${c_reset}"
+    exit 1
+  fi
+
+  git commit -m "Fix formatting / build issues"
```

Comment:
nit: put the tool name in the commit so it's clear it was done by an automated tool.

"Fix formatting / build issues (buildfix.sh)"

Reply:
Done